sklearn/mixture/base.py

```diff
@@ -128,10 +133,10 @@ def _initialize_parameters(self, X):
             resp = np.zeros((n_samples, self.n_components))
             label = cluster.KMeans(n_clusters=self.n_components, n_init=1,
                                    random_state=random_state).fit(X).labels_
-            resp[range(n_samples), label] = 1
+            resp[np.arange(n_samples), label] = 1
```

Comment:
I'm not agree with you. `resp[:, label]` will select the set :
```
[[ resp[0, label[0]], resp[0, label[1]], ... resp[0, label[n]] ],
 [ resp[1, label[0]], resp[1, label[1]], ... resp[1, label[n]] ],
 ...
 [ resp[n, label[0]], resp[n, label[1]], ... resp[n, label[n]] ]]
```
not the set :
```
[ resp[0, label[0]], resp[1, label[1]], ..., resp[n, label[n] ]
```
To select the second set, the only way is to do `resp[np.arange(n_samples), label]`